### DISCUSSION TITLE
How change IEEE(=MAC) in CC2531

### DISCUSSION THREAD

**Discussion by espinardo** [Koenkk/Z-Stack-firmware#486] [Q&A]
Hello everyone!

My CC2531 broke down a while ago and I ordered a new one to replace in my Zigbee2MQTT system.

I flash my new coordinator as before with the firmware: 
https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_Home_1.2/bin

So far everything is correct, I see my device.

The problem comes when I modify my configuration.yaml, I have to do it because the new device has another ID.

/dev/serial/by-id/usb-Texas_Instruments_TI_CC2531_USB_CDC___0X00124B000151B1F4-if00

Next I can start Z2M and see my devices, but apparently these would only pay attention to the old Mac, because I only have error messages when trying to act on a light or device.

https://github.com/Koenkk/Z-Stack-firmware/issues/72
In this thread there are two links that seem to show how to change the MAC of the CC2531, but they are broken:
http://processors.wiki.ti.com/index.php/Z-Stack_FAQ#Device_MAC_Address_in_Z-Stack
https://github.com/andreasbrett/zigbee2mqtt.io/blob/patch-1/information/cc2531_backup_stick.md#how-to-clone-your-stick-windows-only-currently

Has anyone been able to change the MAC so as not to have to register all the devices in Zigbee2MQTT again?

**Comment by Hedda** [comment]
Best post Zigbee2MQTT querstions to -> https://github.com/Koenkk/zigbee2mqtt/discussions but FYI, copying IEEE address in in Z2M FAQ:

https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

https://www.zigbee2mqtt.io/guide/adapters/flashing/copy_ieeaddr.html

Another unofficial alternative would be to use zigpy-cli to backup one and restore to the other as that automatically copies the whole network and the IEEE address, read this discussion:

https://github.com/Koenkk/zigbee2mqtt/discussions/16478

FYI, zigpy-cli and Zigbee2MQTT gateway backups are not compatible even though both use "Open ZigBee Coordinator Backup Format", that is because the Zigbee network backup is only a high-level and stack-agnostic backup of your device's network data (written in the [Open Coordinator Backup Format](https://github.com/zigpy/open-coordinator-backup/)) but it at least allows you to snapshot the device state and move your network between any hardware and firmware versions supported by zigpy-cli. This backup format is human-readable and fully documented so users have to option to manually fill out or change the appropriate information by hand to form a network if they are migrating from a Zigbee Coordinator that is not currently supported.

https://github.com/zigpy/open-coordinator-backup
